src/ol/view.js

```diff
@@ -788,8 +788,9 @@ ol.View.prototype.getState = function() {
 
 
 /**
- * Get the current zoom level. Return undefined if the current
- * resolution is undefined or not within the "resolution constraints".
+ * Get the current zoom level.  If you configured your view with a resolutions
+ * array (this is rare), this method may return non-integer zoom levels (so
+ * the zoom level is not safe to use as an index into a resolutions array).
```

Comment:
Returning the closest zoom level looks desirable to me.

However I always assumed the zoom level was an the index in the resolutions array, an integer.
It looks strange to me to return a non-integer value. Why not return the closest integer?

I am probably missing something.
@fredj, what do you think about this change?

Reply:
I now understand the rationale. With a non-integer value it becomes possible to design a richer zoom level selector, where the user can visually see where he stands between 2 integer / pixel perfect zoom levels. Great.

Reply:
this is what we called fractionalZoom in openlayers 2 http://dev.openlayers.org/examples/fractional-zoom.html

Reply:
Thanks @bartvde for the pointer.

Reply:
Also note that we return non integer values for zoom *before* this change.  The difference here is the behavior when the resolution is higher than the max or lower than the min.  See the tests for detail.

Reply:
I see. Thanks @tschaub for the explanation.